test/e2e/apps/rc.go

```diff
@@ -139,14 +140,27 @@ var _ = SIGDescribe("ReplicationController", func() {
 
 		rcWatchChan := rcWatch.ResultChan()
 
+		ginkgo.By("waiting for RC to be added")
+		eventFound := false
+		for watchEvent := range rcWatchChan {
+			if watchEvent.Type == watch.Added {
+				eventFound = true
+				break
+			}
+		}
+		framework.ExpectEqual(eventFound, true, "failed to find RC %v event", watch.Added)
```

Comment:
nit: for the number of times this is copy-pasted around, I would pull this into a function
```
expectWatchEventType(rcWatch, watch.Added)
```

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/watch.go#L397-L431 has something to crib from

Reply:
Using `framework.WatchUntilWithoutRetry` (introduced in #91416) for these checks as of 313f0341ca3cc3de91c664e6f5d001dab1a6852e